src/Native/System.Native/pal_process.cpp

```diff
@@ -346,3 +356,75 @@ int32_t WTermSig(int32_t status)
 {
     return WTERMSIG(status);
 }
+
+static
+int32_t ConvertPathConfNameToPlatform(PathConfNames& name)
+{
+    switch (name)
+    {
+        case PathConfNames::PAL_PC_LINK_MAX: 
+            return _PC_LINK_MAX;
+        case PathConfNames::PAL_PC_MAX_CANON:
+            return _PC_MAX_CANON;
+        case PathConfNames::PAL_PC_MAX_INPUT:
+            return _PC_MAX_INPUT;
+        case PathConfNames::PAL_PC_NAME_MAX:
+            return _PC_NAME_MAX;
+        case PathConfNames::PAL_PC_PATH_MAX:
+            return _PC_PATH_MAX;
+        case PathConfNames::PAL_PC_PIPE_BUF:
+            return _PC_PIPE_BUF;
+        case PathConfNames::PAL_PC_CHOWN_RESTRICTED:
+            return _PC_CHOWN_RESTRICTED;
+        case PathConfNames::PAL_PC_NO_TRUNC:
+            return _PC_NO_TRUNC;
+        case PathConfNames::PAL_PC_VDISABLE:
+            return _PC_VDISABLE;
+    }
+
+    assert(!"Missing case in PathConfNames conversion");
+    return -1;
+}
+
+extern "C"
+int64_t PathConf(const char* path, PathConfNames name)
+{
+    return pathconf(path, ConvertPathConfNameToPlatform(name));
+}
+
+extern "C"
+int32_t GetMaximumPath()
+{
+    int32_t result = pathconf("/", _PC_PATH_MAX);
+    if (result == -1)
+    {
+        result = PATH_MAX;
+    }
+
+    return result;
+}
+
+extern "C"
+int32_t GetPriority(PriorityWhich which, int32_t who)
+{
+    return getpriority(static_cast<int>(which), who);
+}
+
+extern "C"
+int32_t SetPriority(PriorityWhich which, int32_t who, int32_t nice)
+{
+    return setpriority(static_cast<int>(which), who, nice);
+}
+
+extern "C"
+void GetCwd(FoundWorkingDirectory wdCallback)
+{
+    // The POSIX.1-2001 standard allows passing a nullptr buffer and having the 
+    // OS allocate enough space for the path; use this to get the working directory
+    // without having to allocate and reallocate to get the right size buffer. We 
+    // can then pass the C# code a pointer to the start of the buffer and they can
+    // copy the string to managed code before we free it.
+    char* bufferStart = getcwd(nullptr, 0);
+    wdCallback(bufferStart);
+    free(bufferStart);
+}
```

Comment:
The old managed code had a fast path for stackallocable size. It's probably worth replicating that here.